com.unity.render-pipelines.core/Editor/PostProcess/LensFlareEditor.cs

```diff
@@ -0,0 +1,21 @@
+using System.IO;
+using UnityEditor;
+using UnityEngine;
+
+namespace UnityEngine
+{
+    /// <summary>
+    /// Editor for LensFlare (builtin): Editor to show an error message
+    /// </summary>
+    [CustomEditor(typeof(UnityEngine.Flare))]
```

Comment:
Shouldn't this use the[ CustomEditorForRenderPipelineAttribute](https://docs.unity3d.com/ScriptReference/CustomEditorForRenderPipelineAttribute.html) instead?

Reply:
This is the attribute we use for other cases IES, etc.

Reply:
Doesn't this make the override implicitly rely on the order assemblies are loaded by the engine? Furthermore, using this attribute will override the inspector regardless of what pipeline is active, if any at all. Using the CustomEditorForRenderPipeline attribute will only override the inspector if a RenderPipeline is active.

Reply:
This code is common to HDRP & URP (and all possible SRP), so not specific to a single RenderPipeline.

Reply:
CustomEditorForRenderPipeline is per SRP. we want this for all SRP but not built-in.
So either in this inspector we check if we are in built-in and mimic the built-in inspector.
Or we should find a way to use CustomEditorForRenderPipeline with any SRP different than built-in.

What can work is the following pattern maybe. It require a really small implementation in any CustomSRP but on the other end, it also require this Editor to be public.
```
public class SRPLensFlareEditor : Editor { /* current code */ } //and everything internal in it
```

in HDRP
```
[CustomEditorForRenderPipeline(typeof(UnityEngine.Flare), typeof(HDRenderPipelineAsset))]
class HDRPLensFlareEditor : SRPLensFlareEditor {}
```

in URP
```
[CustomEditorForRenderPipeline(typeof(UnityEngine.Flare), typeof(UniversalRenderPipelineAsset))]
class URPLensFlareEditor : SRPLensFlareEditor {}
```

in CustomRP, user should had this two lines for this to work
```
[CustomEditorForRenderPipeline(typeof(UnityEngine.Flare), typeof(CustomRenderPipelineAsset))]
class CRPLensFlareEditor : SRPLensFlareEditor {}
```

Reply:
I understand. There is multiple scenarios to consider: 
1. Any SRP is active, this Editor should overwrite the built-in editor.
2. No SRP is active, built-in renderer is used. The Editor should not be overwritten.

As it stands having core SRP installed will overwrite the inspector in both cases. If it is possible to use `CustomEditorForRenderPipelineAttribute` we can overwrite it for the common base of all SRPs. This will mean the editor will only be overwritten when an SRP is active.

Reply:
`CustomEditorForRenderPipeline` checks if a class is `IsAssignableFrom()`: 
```
// if an editor targets the current RP directly, use it immediately.
// else use an inherited RP editor as fallback, but continue searching for a direct option.
```

Reply:
Indeed. I quickly looked at the code to check too.

So it seams we can make it work without implementation in each SRP:
```
[CustomEditorForRenderPipeline(typeof(UnityEngine.Flare), typeof(RenderPipelineAsset))]  //base class of all RPAsset -> all SRP, not buit-in
class SRPLensFlareEditor : Editor { /* current code */ }
```

Reply:
Let me know if the change done are compliant with your recommendation.

Reply:
Looks good 👍 

One note; In case of `LensFlareEditorSRP` it should be safe to use the normal attribute as built-in does not have `LensFlareDataSRP`. If I am not mistaken there won't be an editor for `LensFlareDataSRP` now if no SRP asset is active.